src/components/structures/RoomView.tsx

```diff
@@ -528,7 +530,18 @@ export default class RoomView extends React.Component<IProps, IState> {
     }
 
     shouldComponentUpdate(nextProps, nextState) {
-        return (objectHasDiff(this.props, nextProps) || objectHasDiff(this.state, nextState));
+        const hasPropsDiff = objectHasDiff(this.props, nextProps);
+
+        const newUpgradeRecommendation = nextState.upgradeRecommendation || {}
+
+        const state = _.omit(this.state, ['upgradeRecommendation']);
```

Comment:
fwiw, we have our own utility `objectExcluding` to accomplish this.

Reply:
Shall we use the `lodash` implementation instead? It looks like we already go as far as loading the library in the codebase

Their implementation is thoroughly tested and has generally pretty good performance? 
I don't really mind either way

Reply:
I'm not personally opinionated on it: just wanted to mention that we have a thing that does this.